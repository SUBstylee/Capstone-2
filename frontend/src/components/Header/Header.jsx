import { Navbar, Nav, Container,NavDropdown } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { useDispatch,useSelector } from "react-redux";

import brandLogo from "../../assets/images/logos/TAA-header-color-transparent.png";
import { logout } from "../../actions/userActions";

const Header = () => {
  const userLogin=useSelector(state=>state.userLogin);
  const {userInfo}=userLogin;
  const cart = useSelector((state) => state.cart);
  const cartLength = cart.cartItems.length;
  const dispatch=useDispatch();
  const cartColor= cartLength===0?'#fc917e':'#98cef9';

  const logoutHandler=()=>{
    dispatch(logout());
  };

  return (
    <header className='Header'>
      <Navbar bg='light' expand='lg' collapseOnSelect>
        <LinkContainer to='/'>
          <Navbar.Brand>
            <img
              className='ps-2'
              src={brandLogo}
              alt='Totally Awesome Apparel Logo'
              height='36'
            />
          </Navbar.Brand>
        </LinkContainer>
        <Container>
          <Navbar.Toggle aria-controls='basic-navbar-nav' />
          <Navbar.Collapse id='basic-navbar-nav'>
            <Nav className='ms-auto'>
              <LinkContainer to='/cart'>
              <Nav.Link>
                <span className='fa-layers fa-fw'>
                  <i className='fas fa-shopping-cart fa-lg'></i>
                    <span className='fa-layers-counter fa-layers-top-left' style={{ background: `${cartColor}`, borderRadius:'50%', padding:'0 6px', marginRight:'1px'}}>{cartLength}</span>
                </span>
                Cart
              </Nav.Link>
              </LinkContainer>
              {userInfo?(
                <NavDropdown title={userInfo.name} id='username'>
                  <LinkContainer to='/profile'>
                    <NavDropdown.Item>Profile</NavDropdown.Item>  
                  </LinkContainer>
                    <NavDropdown.Item onClick={logoutHandler}>Logout</NavDropdown.Item>
                </NavDropdown>
              ):(
                <LinkContainer to='/login'>
                  <Nav.Link>
                    <i className='fa-solid fa-right-to-bracket'></i> Sign In
                  </Nav.Link>
                </LinkContainer>
              )}
              
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </header>
  );
};

export default Header;
