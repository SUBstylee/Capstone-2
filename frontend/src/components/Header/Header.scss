.Header{
    text-transform: uppercase;
    .Header-cats{
        text-align: center;
        padding: 15px 0;
        height: 50px;
        border-bottom: 2px solid #ededed;
    }
}