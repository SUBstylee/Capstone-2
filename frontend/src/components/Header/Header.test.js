import { render } from '@testing-library/react';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';

test("renders Footer component without crashing", () => {
    const { getByText } = render(
        <BrowserRouter>
            <Header />
        </BrowserRouter>
    );

    expect(getByText("Sign In")).toBeInTheDocument();
    expect(getByText("Cart")).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <BrowserRouter>
            <Header />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});