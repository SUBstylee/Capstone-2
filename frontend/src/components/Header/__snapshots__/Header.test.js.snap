// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        class="active navbar-brand"
        href="/"
      >
        <img
          alt="Totally Awesome Apparel Logo"
          class="ps-2"
          height="36"
          src="TAA-header-color-transparent.png"
        />
      </a>
      <div
        class="container"
      >
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="ms-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/cart"
              href="/cart"
            >
              <i
                class="fa-solid fa-cart-shopping"
              />
               Cart
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/login"
              href="/login"
            >
              <i
                class="fa-solid fa-right-to-bracket"
              />
               Sign In
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
